addons/mail/models/discuss/ir_websocket.py

```diff
@@ -22,6 +22,17 @@ def _get_im_status(self, data):
     @add_guest_to_context
     def _build_bus_channel_list(self, channels):
         channels = list(channels)  # do not alter original list
+        guest_channels = []
+        for channel in channels.copy():
+            if isinstance(channel, str) and channel.startswith("mail.guest_"):
+                guest_channels.append(channel)
+                channels.remove(channel)
+        if guest_channels:
+            new_context = {**self.env.context}
+            new_context.update(
+                guest=self.env["mail.guest"]._get_guest_from_token(guest_channels[0].split("_")[1])
+            )
+            self = self.with_context(new_context)
         guest = self.env["mail.guest"]._get_guest_from_context()
```

Comment:
How does it play when there is already a guest in the context? which one should take priority?

Reply:
Guest passed explicitly via `channels` is kept.

Reply:
Ok but we should keep it only if it is valid then, and probably only the last one given.